docs/webhook.asciidoc

```diff
@@ -0,0 +1,191 @@
+[id="{p}-webhook"]
+== Validating webhook
+
+A validating webhook provides additional validation of Elasticsearch resources beyond what can be specified in OpenAPI specs.
+This allows advanced validation to help prevent Elasticsearch resources that are invalid and will not function correctly
+from being admitted by the Kubernetes API server, which can aid troubleshooting.
+
+[float]
+=== Architecture
+The webhook is composed of 4 main components, here is a brief description of each of them to understand how they interact,
+their naming and how they are managed.
+
+. A `ValidatingWebhookConfiguration` object that defines the validating webhook. In ECK it is named `elastic-es-validation.k8s.elastic.co`. It must be created before to start the operator, the `caBundle` field can be automatically managed as part of the
+automatic certificate management _(see below)_.
+. A Kubernetes Service is used to expose the validating server, in ECK its name is fixed to `elastic-webhook-server`.
+It lives in the same Namespace that the webhook server.
+. A webhook server that actually validates the submitted resources. In ECK it is the operator itself when it
+is configured to act the `webhook` role. See <<{p}-operator-config,Configuring ECK>> if want to learn more about the
```

Comment:
...when it is configured to **have** the `webhook` role...
... if **you** want to learn...